internal/datasystem/fdv1_datasystem.go

```diff
@@ -0,0 +1,146 @@
+	if err != nil {
+		return nil, err
+	}
+	system.dataStore = store
+
+	system.dataStoreStatusProvider = datastore.NewDataStoreStatusProviderImpl(store, dataStoreUpdateSink)
+
+	dataSourceUpdateSink := datasource.NewDataSourceUpdateSinkImpl(
+		store,
+		system.dataStoreStatusProvider,
+		system.dataSourceStatusBroadcaster,
+		system.flagChangeEventBroadcaster,
+		clientContext.GetLogging().LogDataSourceOutageAsErrorAfter,
+		clientContext.GetLogging().Loggers,
+	)
+
+	dataSource, err := createDataSource(clientContext, dataSourceFactory, dataSourceUpdateSink)
+	if err != nil {
+		return nil, err
+	}
+	system.dataSource = dataSource
+	system.dataSourceStatusProvider = datasource.NewDataSourceStatusProviderImpl(
+		system.dataSourceStatusBroadcaster,
+		dataSourceUpdateSink,
+	)
+
+	return system, nil
+
+}
+
+func createDataSource(
+	context *internal.ClientContextImpl,
+	dataSourceBuilder subsystems.ComponentConfigurer[subsystems.DataSource],
+	dataSourceUpdateSink subsystems.DataSourceUpdateSink,
+) (subsystems.DataSource, error) {
+	if context.Offline {
+		context.GetLogging().Loggers.Info("Starting LaunchDarkly client in offline mode")
+		dataSourceUpdateSink.UpdateStatus(interfaces.DataSourceStateValid, interfaces.DataSourceErrorInfo{})
+		return datasource.NewNullDataSource(), nil
+	}
+	factory := dataSourceBuilder
+	if factory == nil {
+		// COVERAGE: can't cause this condition in unit tests because it would try to connect to production LD
+		factory = ldcomponents.StreamingDataSource()
+	}
+	contextCopy := *context
+	contextCopy.BasicClientContext.DataSourceUpdateSink = dataSourceUpdateSink
+	return factory.Build(&contextCopy)
+}
+
+func (f *FDv1) DataSourceStatusBroadcaster() *internal.Broadcaster[interfaces.DataSourceStatus] {
+	return f.dataSourceStatusBroadcaster
+}
+
+func (f *FDv1) DataSourceStatusProvider() interfaces.DataSourceStatusProvider {
+	return f.dataSourceStatusProvider
+}
+
+func (f *FDv1) DataStoreStatusBroadcaster() *internal.Broadcaster[interfaces.DataStoreStatus] {
+	return f.dataStoreStatusBroadcaster
+}
+
+func (f *FDv1) DataStoreStatusProvider() interfaces.DataStoreStatusProvider {
+	return f.dataStoreStatusProvider
+}
+
+func (f *FDv1) FlagChangeEventBroadcaster() *internal.Broadcaster[interfaces.FlagChangeEvent] {
+	return f.flagChangeEventBroadcaster
+}
+
+func (f *FDv1) Start(closeWhenReady chan struct{}) {
+	f.dataSource.Start(closeWhenReady)
+}
+
+func (f *FDv1) Stop() error {
+	if f.dataSource != nil {
+		_ = f.dataSource.Close()
+	}
+	if f.dataStore != nil {
+		_ = f.dataStore.Close()
+	}
+	if f.dataSourceStatusBroadcaster != nil {
+		f.dataSourceStatusBroadcaster.Close()
+	}
+	if f.dataStoreStatusBroadcaster != nil {
+		f.dataStoreStatusBroadcaster.Close()
+	}
+	if f.flagChangeEventBroadcaster != nil {
+		f.flagChangeEventBroadcaster.Close()
+	}
+	return nil
+}
+
+func (f *FDv1) Offline() bool {
+	return f.dataSource == datasource.NewNullDataSource()
+}
+
```

Comment:
Here's a new DataSystem method: the "data status"!


This gives users what they really want to know: does the SDK have data, and if so, is it "fresh".